cmd/up.go

```diff
@@ -224,59 +219,55 @@ func clusterUpLocal(ctx *cli.Context) error {
 
 func clusterUpDind(ctx *cli.Context) error {
 	// get dind config
-	rkeConfig, disablePortCheck, dindSubnet, err := getDindConfig(ctx)
+	rkeConfig, disablePortCheck, err := getDindConfig(ctx)
 	if err != nil {
 		return err
 	}
 	// setup dind environment
-	if err = createDINDEnv(context.Background(), dindSubnet, rkeConfig); err != nil {
+	if err = createDINDEnv(context.Background(), rkeConfig); err != nil {
 		return err
 	}
 	// start cluster
 	_, _, _, _, _, err = ClusterUp(context.Background(), rkeConfig, hosts.DindConnFactory, hosts.DindHealthcheckConnFactory, nil, false, "", false, disablePortCheck)
 	return err
 }
 
-func getDindConfig(ctx *cli.Context) (*v3.RancherKubernetesEngineConfig, bool, string, error) {
+func getDindConfig(ctx *cli.Context) (*v3.RancherKubernetesEngineConfig, bool, error) {
 	disablePortCheck := ctx.Bool("disable-port-check")
-	dindSubnet := ctx.String("dind-subnet")
 	clusterFile, filePath, err := resolveClusterFile(ctx)
 	if err != nil {
-		return nil, disablePortCheck, dindSubnet, fmt.Errorf("Failed to resolve cluster file: %v", err)
+		return nil, disablePortCheck, fmt.Errorf("Failed to resolve cluster file: %v", err)
 	}
 	clusterFilePath = filePath
 
 	rkeConfig, err := cluster.ParseConfig(clusterFile)
 	if err != nil {
-		return nil, disablePortCheck, dindSubnet, fmt.Errorf("Failed to parse cluster file: %v", err)
+		return nil, disablePortCheck, fmt.Errorf("Failed to parse cluster file: %v", err)
 	}
 
 	rkeConfig, err = setOptionsFromCLI(ctx, rkeConfig)
 	if err != nil {
-		return nil, disablePortCheck, dindSubnet, err
+		return nil, disablePortCheck, err
 	}
 	// Setting conntrack max for kubeproxy to 0
 	if rkeConfig.Services.Kubeproxy.ExtraArgs == nil {
 		rkeConfig.Services.Kubeproxy.ExtraArgs = make(map[string]string)
 	}
 	rkeConfig.Services.Kubeproxy.ExtraArgs["conntrack-max-per-core"] = "0"
 
-	return rkeConfig, disablePortCheck, dindSubnet, nil
+	return rkeConfig, disablePortCheck, nil
 }
 
-func createDINDEnv(ctx context.Context, dindSubnet string, rkeConfig *v3.RancherKubernetesEngineConfig) error {
-	if dindSubnet == "" {
-		logrus.Infof("[%s] dind subnet didn't get specified, using default subnet [%s]", dind.DINDPlane, dind.DINDSubnet)
-		dindSubnet = dind.DINDSubnet
-	}
-	if err := dind.CreateDindNetwork(ctx, dindSubnet); err != nil {
-		return fmt.Errorf("Failed to create dind network: %v", err)
-	}
-
-	for _, node := range rkeConfig.Nodes {
-		if err := dind.StartUpDindContainer(ctx, node.Address, dind.DINDNetwork); err != nil {
+func createDINDEnv(ctx context.Context, rkeConfig *v3.RancherKubernetesEngineConfig) error {
+	for i := range rkeConfig.Nodes {
+		address, err := dind.StartUpDindContainer(ctx, rkeConfig.Nodes[i].Address, dind.DINDNetwork)
+		if err != nil {
 			return err
 		}
+		if rkeConfig.Nodes[i].HostnameOverride == "" {
```

Comment:
that's already will be checked in the validate cluster, this was added to be able to use the address as node name instead of the ip assigned to the dind container